ios/jailed/patchapp.sh

```diff
@@ -0,0 +1,377 @@
+#!/bin/bash
+
+#
+# You might need to change these
+#
+OPTOOL="/usr/local/bin/optool"
+IOSDEPLOY="/usr/local/bin/ios-deploy"
+
+#
+# You shouldn't need to change these unless you have multiple Dev certs
+#
+COMMAND=$1
+IPA=$2
+MOBILEPROVISION=$3
+DEV_CERT_NAME="iPhone Developer"
+CODESIGN_NAME=`security dump-keychain login.keychain|grep "$DEV_CERT_NAME"|head -n1|cut -f4 -d \"|cut -f1 -d\"`
+TMPDIR=".patchapp.cache"
+DYLIB=".theos/obj/@@PROJECTNAME@@.dylib"
+
+#
+# Usage / syntax
+#
+function usage {
+	if [ "$2" == "" -o "$1" == "" ]; then
+		cat <<USAGE
+Usage: $0 <command> file.ipa [file.mobileprovision | your.bundle.id | -w]
+Where 'command' is one of:
+	info    - Show the information required to create a Provisioning Profile
+	          that matches the specified .ipa file
+	patch   - Inject the tweak into the specified .ipa file.
+	    	  Requires that you specify a .mobileprovision file
+	install - Patch and deploy the app onto your device via USB
+   	debug   - Like deploy, but also debug the app with LLDB
+USAGE
+	fi
+}
+
+#
+# Setup all the things.
+#
+function setup_environment {
+	if [ "$IPA" == "" ]; then
+		usage
+		exit 1
+	fi
+	if [ ! -r "$IPA" ]; then
+		echo "$IPA not found or not readable"
+		exit 1
+	fi
+
+	# setup
+	rm -rf "$TMPDIR" >/dev/null 2>&1
+	mkdir "$TMPDIR"
+	SAVED_PATH=`pwd`
+
+	# uncompress the IPA into tmpdir
+	echo '[+] Unpacking the .ipa file ('"`pwd`/$IPA"')...'
+	unzip -o -d "$TMPDIR" "$IPA" >/dev/null 2>&1
+	if [ "$?" != "0" ]; then
+		echo "Couldn't unzip the IPA file."
+		exit 1
+	fi
+
+	cd "$TMPDIR"
+	cd Payload/*.app
+	if [ "$?" != "0" ]; then
+		echo "Couldn't change into Payload folder. Wat."
+		exit 1
+	fi
+	APP=`pwd`
+	APP=${APP##*/}
+	APPDIR=$TMPDIR/Payload/$APP
+	cd "$SAVED_PATH"
+	BUNDLE_ID=`plutil -convert xml1 -o - $APPDIR/Info.plist|grep -A1 CFBundleIdentifier|tail -n1|cut -f2 -d\>|cut -f1 -d\<`-patched
+	APP_BINARY=`plutil -convert xml1 -o - $APPDIR/Info.plist|grep -A1 Exec|tail -n1|cut -f2 -d\>|cut -f1 -d\<`
```

Comment:
Use [defaults or PlistBuddy](http://stackoverflow.com/questions/4328501/how-to-read-plist-information-bundle-id-from-a-shell-script) here.